### PR TITLE
[SPARK-42483][TESTS] Regenerate benchmark results

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This aims to regenerate benchmark results on `master` branch as a baseline for Spark 3.5.0 and a way to comparing Apache Spark 3.4.0 branch.

### Why are the changes needed?

These are reference values with minor changes.
```
- OpenJDK 64-Bit Server VM 1.8.0_352-b08 on Linux 5.15.0-1023-azure
+ OpenJDK 64-Bit Server VM 1.8.0_362-b09 on Linux 5.15.0-1031-azure

- OpenJDK 64-Bit Server VM 11.0.17+8 on Linux 5.15.0-1023-azure
+ OpenJDK 64-Bit Server VM 11.0.18+10 on Linux 5.15.0-1031-azure

- OpenJDK 64-Bit Server VM 17.0.5+8 on Linux 5.15.0-1023-azure
+ OpenJDK 64-Bit Server VM 17.0.6+10 on Linux 5.15.0-1031-azure
```

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Manual review.

### CODE REVIEW

File: core/benchmarks/ZStandardBenchmark-jdk11-results.txt [hunk 1]
```diff
@@ -2,26 +2,26 @@
...
-Compression 10000 times at level 3 without buffer pool           1137           1138           2          0.0      113664.6       0.8X
-Compression 10000 times at level 1 with buffer pool               662            664           1          0.0       66244.7       1.3X
-Compression 10000 times at level 2 with buffer pool               725            726           1          0.0       72541.4       1.2X
-Compression 10000 times at level 3 with buffer pool               929            930           2          0.0       92851.4       0.9X
+Compression 10000 times at level 1 without buffer pool            605            812         220          0.0       60521.0       1.0X
+Compression 10000 times at level 2 without buffer pool            665            678          20          0.0       66512.5       0.9X
+Compression 10000 times at level 3 without buffer pool            890            903          20          0.0       88961.3       0.7X
+Compression 10000 times at level 1 with buffer pool               829            839          11          0.0       82940.2       0.7X
```
Comment (dongjoon-hyun) [inline core/benchmarks/ZStandardBenchmark-jdk11-results.txt:12]: I'll take a look at this after this PR.

Comment (dongjoon-hyun) [reply core/benchmarks/ZStandardBenchmark-jdk11-results.txt:12]: Java 8/17 doesn't have this regression.

File: sql/catalyst/benchmarks/EnumTypeSetBenchmark-jdk11-results.txt [hunk 2]
```diff
@@ -1,105 +1,105 @@
-OpenJDK 64-Bit Server VM 11.0.17+8 on Linux 5.15.0-1023-azure
-Intel(R) Xeon(R) Platinum 8272CL CPU @ 2.60GHz
+OpenJDK 64-Bit Server VM 11.0.18+10 on Linux 5.15.0-1031-azure
+Intel(R) Xeon(R) Platinum 8370C CPU @ 2.80GHz
 Test contains use empty Set:              Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
 ------------------------------------------------------------------------------------------------------------------------
-Use HashSet                                           4              4           0        226.9           4.4       1.0X
-Use EnumSet                                           1              1           0        737.3           1.4       3.2X
+Use HashSet                                           0              1           0       2440.2           0.4       1.0X
+Use EnumSet                                           1              1           0        884.8           1.1       0.4X
```
Comment (dongjoon-hyun) [inline sql/catalyst/benchmarks/EnumTypeSetBenchmark-jdk11-results.txt:6]: We need to investigate this reversed ratio.

Comment (dongjoon-hyun) [reply sql/catalyst/benchmarks/EnumTypeSetBenchmark-jdk11-results.txt:6]: `HashSet` seems to get some improvements.

File: sql/catalyst/benchmarks/EnumTypeSetBenchmark-results.txt [hunk 3]
```diff
@@ -1,105 +1,105 @@
-OpenJDK 64-Bit Server VM 1.8.0_352-b08 on Linux 5.15.0-1023-azure
+OpenJDK 64-Bit Server VM 1.8.0_362-b09 on Linux 5.15.0-1031-azure
 Intel(R) Xeon(R) Platinum 8370C CPU @ 2.80GHz
 Test contains use empty Set:              Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
 ------------------------------------------------------------------------------------------------------------------------
-Use HashSet                                           5              5           0        209.4           4.8       1.0X
-Use EnumSet                                           2              2           0        459.8           2.2       2.2X
+Use HashSet                                           1              1           1       1972.0           0.5       1.0X
+Use EnumSet                                           2              2           0        444.0           2.3       0.2X
```
Comment (dongjoon-hyun) [inline sql/catalyst/benchmarks/EnumTypeSetBenchmark-results.txt:6]: ditto.

File: sql/catalyst/benchmarks/HashBenchmark-jdk11-results.txt [hunk 4]
```diff
@@ -2,69 +2,69 @@
...
 ------------------------------------------------------------------------------------------------------------------------
-interpreted version                                3763           3769           8        142.7           7.0       1.0X
-codegen version                                    4658           4662           5        115.3           8.7       0.8X
-codegen version 64-bit                             4706           4710           6        114.1           8.8       0.8X
-codegen HiveHash version                           3998           3998           0        134.3           7.4       0.9X
+interpreted version                                4933           4935           2        108.8           9.2       1.0X
+codegen version                                    5135           5141           9        104.6           9.6       1.0X
+codegen version 64-bit                             5071           5079          10        105.9           9.4       1.0X
+codegen HiveHash version                           4326           4326           0        124.1           8.1       1.1X
```
Comment (dongjoon-hyun) [inline sql/catalyst/benchmarks/HashBenchmark-jdk11-results.txt:12]: Now, this is the fastest.

File: sql/core/benchmarks/UpdateFieldsBenchmark-results.txt [hunk 5]
```diff
@@ -2,25 +2,25 @@
...
 
-OpenJDK 64-Bit Server VM 1.8.0_352-b08 on Linux 5.15.0-1023-azure
+OpenJDK 64-Bit Server VM 1.8.0_362-b09 on Linux 5.15.0-1031-azure
 Intel(R) Xeon(R) CPU E5-2673 v4 @ 2.30GHz
 Add 50 columns and drop 50 columns at 100 different depths of nesting:  Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
 -----------------------------------------------------------------------------------------------------------------------------------------------------
-To non-nullable StructTypes using performant method                             5520           5639         168          0.0      Infinity       1.0X
-To nullable StructTypes using performant method                                 2657           2708          72          0.0      Infinity       2.1X
+To non-nullable StructTypes using performant method                             3126           3150          34          0.0      Infinity       1.0X
+To nullable StructTypes using performant method                                 3136           4768        2309          0.0      Infinity       1.0X
```
Comment (dongjoon-hyun) [inline sql/core/benchmarks/UpdateFieldsBenchmark-results.txt:24]: This looks like a regression. We need to take a look at this later.

File: sql/core/benchmarks/TPCDSQueryBenchmark-jdk11-results.txt [hunk 6]
```diff
@@ -1,810 +1,810 @@
...
-q2                                                 1686           1696          15          1.3         755.2       1.0X
+q2                                                 1585           1899         444          1.4         710.1       1.0X
 
-OpenJDK 64-Bit Server VM 11.0.17+8 on Linux 5.15.0-1023-azure
+OpenJDK 64-Bit Server VM 11.0.18+10 on Linux 5.15.0-1031-azure
 Intel(R) Xeon(R) CPU E5-2673 v3 @ 2.40GHz
 TPCDS Snappy:                             Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
 ------------------------------------------------------------------------------------------------------------------------
-q3                                                  718            759          41          4.1         241.8       1.0X
+q3                                                  996           1035          55          3.0         335.3       1.0X
```
Comment (dongjoon-hyun) [inline sql/core/benchmarks/TPCDSQueryBenchmark-jdk11-results.txt:17]: Maybe, slower?

File: sql/core/benchmarks/SortBenchmark-jdk17-results.txt [hunk 7]
```diff
@@ -2,15 +2,15 @@
...
-radix sort two bytes                                373            375           2         66.9          14.9      32.3X
-radix sort eight bytes                             1415           1417           4         17.7          56.6       8.5X
-radix sort key prefix array                        1930           1966          51         13.0          77.2       6.2X
+reference TimSort key prefix array                12111          12128          23          2.1         484.4       1.0X
+reference Arrays.sort                              2861           2885          35          8.7         114.4       4.2X
+radix sort one byte                                 197            197           0        127.0           7.9      61.5X
+radix sort two bytes                                371            372           0         67.4          14.8      32.6X
+radix sort eight bytes                             1391           1397           8         18.0          55.7       8.7X
+radix sort key prefix array                        1914           1951          52         13.1          76.6       6.3X
```
Comment (dongjoon-hyun) [inline sql/core/benchmarks/SortBenchmark-jdk17-results.txt:14]: In this benchmark, all Java 17 results are faster than Java 8.

File: sql/core/benchmarks/DataSourceReadBenchmark-results.txt [hunk 8]
```diff
@@ -2,430 +2,430 @@
...
-SQL Json                                           7948           7990          60          2.0         505.3       1.3X
-SQL Parquet Vectorized: DataPageV1                  126            149          22        125.2           8.0      83.0X
-SQL Parquet Vectorized: DataPageV2                   99            113          17        158.6           6.3     105.2X
-SQL Parquet MR: DataPageV1                         1777           1784           9          8.8         113.0       5.9X
-SQL Parquet MR: DataPageV2                         1579           1583           6         10.0         100.4       6.6X
-SQL ORC Vectorized                                  158            165           5         99.7          10.0      66.1X
-SQL ORC MR                                         1654           1661           9          9.5         105.2       6.3X
-
-OpenJDK 64-Bit Server VM 1.8.0_352-b08 on Linux 5.15.0-1023-azure
+SQL CSV                                           13143          13363         311          1.2         835.6       1.0X
```
Comment (dongjoon-hyun) [inline sql/core/benchmarks/DataSourceReadBenchmark-results.txt:9]: CSV seems to become 30% slower.

**Comment by dongjoon-hyun** [comment]
When you have some time, could you review this, @viirya ? I want to merge this to proceed the further investigations.

**Review by viirya — APPROVED** [review]

File: sql/core/benchmarks/DataSourceReadBenchmark-results.txt (hunk 8 above)
Comment (viirya) [reply sql/core/benchmarks/DataSourceReadBenchmark-results.txt:9]: Hmm, it's significant.

**Comment by dongjoon-hyun** [comment]
Thank you so much always for your help, @viirya !
Merged to master for Apache Spark 3.5.

**Pull request closed without merging**